spec/unit/util/windows/eventlog_spec.rb

```diff
@@ -0,0 +1,93 @@
+#!/usr/bin/env ruby
+
+require 'spec_helper'
+
+require 'puppet/util/windows'
+
+describe Puppet::Util::Windows::EventLog, :if => Puppet.features.microsoft_windows? do
+
+  before(:each) { @event_log = Puppet::Util::Windows::EventLog.new }
+  after(:each) { @event_log.close }
+
+  describe "self.open" do
+    it "sets a handle to the event log" do
+      default_name = Puppet::Util::Windows::String.wide_string('Puppet')
+      # return nil explicitly just to reinforce that we're not leaking eventlog handle
+      Puppet::Util::Windows::EventLog.any_instance.expects(:RegisterEventSourceW).with(nil, default_name).returns(nil)
+      Puppet::Util::Windows::EventLog.new
+    end
+
+    it "raises an exception if the event log handle is not opened" do
+      # RegisterEventSourceW will return NULL on failure
+      # Stubbing prevents leaking eventlog handle
+      Puppet::Util::Windows::EventLog.any_instance.stubs(:RegisterEventSourceW).returns(FFI::Pointer::NULL_HANDLE)
```

Comment:
This is really minor nit, but `FFI::Pointer::NULL_HANDLE` ... or our embedded `NULL_HANDLE`?